### ISSUE TITLE
Wrap fileHeader related formatting options in Formatting.fileHeader prop

### ISSUE THREAD

**Issue by jorenbroekema** [amzn/style-dictionary#1312]
Right now it's a bit awkward passing the fileHeader specific "formatting" options.
There's a formatting property called "prefix" but that's also an option within the fileHeader options. Same for lineSeparator.
It would make sense for these options not to get magically merged/reused and to be scoped appropriately.

**Comment by DarioSoller** [comment]
I have the use case, where I need to adjust the `header` value of the `formatting` options of the fileHeader. But the `header` sometimes gets overruled by the `commentStyle` fileHeaderCommentStyle property. It's not exactly the topic you originally mentioned above, but might be another thing, which could be improved for v6.0 maybe?

My use case is, that I need to provide legal header comments, that withstand any minification process.

The good thing for now would be, that I could achieve this functionality with an one liner. If we put the following line to [fileHeader function in L56](https://github.com/style-dictionary/style-dictionary/blob/44322842d9e9d9bb346f8f84254b90ae579c2bce/lib/common/formatHelpers/fileHeader.js#L56):
```
commentStyle = file?.options?.commentStyle || options?.commentStyle || commentStyle;
```
we should be able to override the commentStyle for the fileHeader either on file level, or on platform level via the platform options.

In my case I need that to add legal header comments to the asset files, which are save of being removed by any minification process.

I can contribute this, if you like. Should I open a PR for this, @jorenbroekema !?

**Comment by jorenbroekema** [comment]
These issues are pretty old so I've grown to be a bit unfamiliar with this and fileHeaders in general. 

Could you outline the issue with the current API, an example of the output you'd like, and how you would change it (if at all)?

I'd love it if someone could take a look at the weirdness for fileHeaders and how they're configured/getting options passed to them, so definitely open for PRs, I just don't really understand the problem well at the moment 😓 .

**Comment by DarioSoller** [comment]
### Currently
Indeed some weirdness involved here, also with the partly redundant `commentStyles` and `fileHeaderCommentStyles` values that there exist. Right now when you add `commentStyle: commentStyles.long` to a platform config it will be overridden for example for a SCSS file, even though it has the `commentStyle: commentStyles.long` long option set on file level.

### Proposed Solution
With my suggested fix for L56 in the `fileHeader.js` file, this would be solved in a way, that file level commentStyles are actually applied for fileHeaders, and platform level commentStyles are also fixing the currently buggy behavior of commentStyles for SCSS files.

So imagine you have a platform config like this one:
```typescript
import { PlatformConfig } from 'style-dictionary';
import { commentStyles, formats, transforms } from 'style-dictionary/enums';

const webPlatformConfig: PlatformConfig = {
  transforms: [transforms.nameCamel],
  buildPath: "./path/to/some/ouputfile",
  files: [
    {
      destination: "web/tokens.css",
      format: formats.cssVariables,
    },
    {
      destination: "web/tokens.scss",
      format: formats.scssVariables,
    },
    {
      destination: "web/tokens.js",
      format: formats.javascriptEs6,
      options: {
        commentStyle: commentStyles.short
      },
    },
  ],
  options: {
    fileHeader: "myFileHeader",
    commentStyle: commentStyles.long,
    formatting: {
      header: `/*!\n`,
    },
  },
};
```
Then the `commentStyle` of the platform config causes the SCSS file finally to have a long comment (/**...*/) as intuitively expected and the javascript file having a short comment style, because it was overriding the platform config on file level. CSS only can have block comments, so this actually is a bit irrelevant here, therefore just mentioning it as a reminder.

In my case I would also give the Javascript file the long `commentStyle`, because I obviously want the legal comment also to be preserved in a minified Javascript token asset file.

Maybe I should also mention, that I am not using the `registerFileHeader()` function but instead assign my fileHeader inline under the hooks of the SD config. Idk if this makes any difference here!?

### File Header Output Goal
To my knowledge most compilers support a way to preserve certain comments somehow, mostly used for legal purposes.
```scss
/*!
 * This comment will be included even in compressed or minified mode.
 */
```

### Contribution

So could you imagine to accept my one liner fix before v6.0 and a most likely breaking clean up refactor of the commentStyle and fileHeader options for v6.0? 

### My Specs
- `"style-dictionary": "^5.1.1"`
- `"@tokens-studio/sd-transforms": "^2.0.3"`

**Comment by DarioSoller** [comment]
Quick [demo config](https://configurator.tokens.studio/#project=5VtZb+s2E/0r+ny/ogtiWyslpS9FUXR5K+59KIq6KLgMbSWyaIhy6jTwfy+12JZky7KUEGib5CExSZ1DDudoODT5MpHZlkVi9iBFMrmfvCwSw1hMWLSGREYiWUzujaIsL5UUx1ArycuecLwtyhYTezG5O1Vkz5uqXEZ/RclyMSnr9ncHuJ3sxnI3u4Foct2N9nIcz2wn98ZXRq2gGNR+INma3UQm129BFi9vIluztyDbxTeRxctXkOV/SkZVtcG0qKv52TXHaE5lF98BdKSTqHkbinyjR6hJGop84/SrGRmKfONc7+Lh1tjGWTQ9wt0klGGMtVoGkqbRJqteV4vJr2JrUJwYK/wERtGVTQxGwSyNKDGwIRWGKqrQjEw8QjIzPgJmxlqkYIjEyFYg4d5YZdlG3s/nTFA5K9rJWfnKnOMnHMWYxDAty+cVXPXxQ80I02bVRSUQkTJIP2IWbWXztXvFad0OOzXBhrhUMArxmitZaBTkbT5UxyjdqFHSrYom+Zu4U1qAvZU3lR2clqAHnyqZptNDN0r7XHYoKmKRNl2JxJg+dlvzg1n8dFiswDufpz9XUXZlhj7w4mcY5jLFzy1IK+9XreSMxueY0xrNFaITVd7I7kEGxm3uj0F2+pBtCLg5BtntQaaEeTAK2etBxiYGOgoZ9SD7VmCGaAyy34PsYs9DwRjkoAfZZo7vjkIOe5AtbJv27f7cVlAKbJiAXsrXxUw9OFM+sO9jbtT+H3ZZGbtli6MIYGWKcYh9rfriXS9YxJ8vVDWp42i5UjwN8gtDMWfuxSZ5EK6arGR8Ml7DhGcf9iPeGf8pYzq6jem8I2Pauo3pviNjWrqN2RcMuYfUr4Zg+G+YJobTx3/ELPnvx5baXx/B+7Gl9rgWvh9b6llwtRe1IsXJEoYmhlwlhqMSlr7EkAMQOmrx35cYcsKcgOlIDDnCzBsVsbzeNDkIHaQhFn5gDBF71Az2JYbU9JBt6UgMQ+ra45D7EkOfOK4FoxPDZ4hj8edwDXFdGuKUcC0aUqoPQj0aAtMDLRqiNHSJFg2hUO026dAQ8f3Q4lo05HvYQlo05Lq2ZY7W0DIFSIZKyFQScnVIiCKOmKdDQiEGRFwdEkIBc0JLh4TcgBA/0CEhJ1DeGOqQkM0Dz8Na9id95LuhDgnZtuc6bLSEMsDxQAUBypdyOhREbO4B1qGgwALEtAQhlzOLalnIOQGxMdWiICv0Q0+LgqiPfaJFQYEHSMtCznZc8OzRCiLnX4z2KojwgHMtCgJwuJZUKDQp43pikEMcYFpikB2GYOlRUGBTLcs4myBKTC0Kop4d2FoUhF1nwPZrW0FRwqKlGK4hV4+GqMMwBx0awg4ZGTn7NOTzkHItUQghH8bF5D4NeRj5LNChIZd6FtGSCrkq5w9NHRpyqIMQGa2hzTbdxIO35DD39GgIQhZwpkNDag+K6tEQ8UN3XITr01DIfaxHQ4HpYaZlJacSTkS1bMl5nkNDrENDrus4Ph6voSh5HKogrhTk61AQB+aDrUNBnKglBuhQEEeB2hXVs6mNXKxnU9tTC2aqQ0EkcOxAy0ou9G2EmA4F+abteuFQBeV/DqcdRX7CNqt/o7WYnG9014+lmp91HEo8QrWFeu2QtzccbhUtV92A4c2AdTNwkWTf43UUR9A8+rkCzJoH7Vt8PyUZpB2MTdT2OIhgz924HwURmRgCXB9PHCXwI+QHQgYOx+qe3wbmsMFY7iDUxkggUwb+dOG+AwOOt3HWzdp1IrcF2R5KlNAUsIQrXmt1u20POINe8Kk3DLturQ1O8TLFm9Ulg62sblKn6xrSOeKZHu1uWBsNhW2r8pduL/4Iy22M0ysaOjToFNEvXf5cUXwr4iszVdYOB8/ForPzOf6b9rw9KZ+iv9rvySu+9XJ8ZLay8wtQ1sz2rrBX6EP8rE7hvILCuY3CfQWFexuF9woK7zYK9AoKdBNF7ogFyXCGnoCChiNevcV23mdzFowwzNVLeJdI0O0kdRXCDq/ziyxyhVljuVZjrPVidykiPpeFzctQan88Paz0GhVyowIXu4RTLTPzinRJ8BfmXfE7c7/sGFqUJCqMVT1vW7B+12hXb6TGv5/cTVJYRjIPg999v01oft9Izh7k5H4SrTcizYxP2XMM30VFDU6fDZ6KtfG5zEun7Fj8+deLpHrgxThAGvuq9TflEbBpdclIsmmW4kRyka5l/uQimc+NRumdYSfMyKPaGnIkVWiI4jaULG5BETAwYyrs509+AjDyu0ynm03LKFttyYyK9fwK9SI59PSL1ii/VJ1StqEi4dGydStZim1anif77SiF0+3l0ra/H1ZborlkeIKUCFklCLUlV2PNkcImFRSkrO5NHWkaQ2kSbWKclWOq00kq2/o5jv6HVGw3ZTdauHcXWtc7UlYlambmj0AwOWVEv9cdfxvF7GecrSrnyz/OVX/mDQIexdDGrqVn5RJPZlGCTzfh/njCaZTfUZOzcoB3zQfy/uKsbJs3mB/b18/+7e/GME5JLOjjcN5W08qVW8noQf7rNah3Ve6R1TpVnBZ0Vedr47hk/Mv43diNhLccR9bOa8rKfCEH1dtp/tX/FoukYdTDv2e7Sw+vc8SL3vTwBs508qWH6zP6gJ9weSVzDhJdHPQ/cgaOr/r93w==) showing the weird buggy behavior of the commentStyle setting.

### Findings

1. The `commentStyle` setting seems to have no effect, neither on file level nor on platform level.
2. The `commentStyle` setting seems to have no effect, neither on the fileHeader nor on the actual token description comments!?
3. Default header commentStyles differ for JS (long) vs. SCSS (short) for example!?

So I might do something wrong or `commentStyle` is currently buggy?

**Comment by jorenbroekema** [comment]
I think the problem is that you're putting the commentStyle as a direct option rather than wrapping it in formatting option prop:

```js
{
  options: {
    formatting: {
      commentStyle: 'long'
    }
  }
}
```

Has the desired effect for me in your demo, either on the platform level or on the file level.

Understandable because it's poorly documented and slightly confusing the way it's designed and the way these "deep" options can propagate from the platform/file level to eventually the `createPropertyFormatter` format utility/helper.

- https://styledictionary.com/reference/utils/format-helpers/#createpropertyformatter -> most accurate docs I can find on the "formatting" prop which is passed to this format helper by the format itself (the CSS, SCSS, LESS, STYLUS ones)
- https://styledictionary.com/reference/config/#file -> It should be clarified here but it seems to be missing the formatting options.
- https://styledictionary.com/reference/hooks/formats/predefined/#cssvariables kinda documented on the format level but seems to be missing props (?)

PRs definitely welcome for aligning those docs, but I think we should also probably take a look at a potential redesign to avoid this confusion, issues to start a discussion of a better way are welcome as well for a future major update of StyleDictionary.

I also want to create auto-generated typescript docs from the types folder we have, that would help with finding out about these options/props just from navigating the types via anchors on the docs site.

**Comment by jorenbroekema** [comment]
I also understand a bit better now this issue and why I raised it long time ago; there's overlapping formatting properties for both the fileHeader and the format itself, for CSS format you can pass `File.options.formatting.lineSeparator` which is then consumed by both the fileHeader as well as the format, so if you pass 2 newlines for this you would get something like:

```css
/**

 * File Header (lineSeparator consumed here via fileHeader)

 */

:root { /* and here via the format itself */

  --foo: 4px;

  --bar: 8px;

}
```

Although I did notice a bug there where commentStyle option isn't consumed properly by the fileHeader when possible, which I'll fix coz it's a small/easy one...

So basically the flaw in the API design imo is that you're passing options via an interface which is then consumed in 2 different contexts where the overlapping options can have different meanings. I'd rather you pass the fileHeader options as `File.options.fileHeader.formatting` separately, but I think it'd be good to have a more holistic discussion on what's actually a good API design for passing all these possible configuration options via global/platform/file level...

**Comment by DarioSoller** [comment]
I tried it with `options.formatting.commentStyle` with the v5.1.4 version, but that did not work for me. I will try this with your fix in v5.2.0 again.

UPDATE: I also tried it with v5.2.0 but I can't reproduce it locally. Also I can't reproduce it with your suggestion in my [demo](https://configurator.tokens.studio/#project=5VtZb+u4Ff4rqm+LWRDbWikpfSnawXT6Vtz7MCjqQcHl0FYii4YoZ5wJ/N9LLbYlxbKWhJglyYNhkvo+nsPz6ZA0+TKT2Z5FYvEgRTK7n72sEsNYzVi0hURGIlnN7o2iLC+VFMdQK8nLnnC8L8pWM3s1u7tUZM+7qlxGv0TJejUr6453J7iD7MZyd4eRaHLbjfZytmdxkEfjW6NWUBh1HEm2ZYPI5PY9yOL1ILItew+yQzyILF6/gSz/KBlV1Q7Toq4WZ7cCozmUXXwn0IlBosZtLPLAiFCDNBZ54PCrERmLPHCsD/F4b+zjLJqf4QYJZRxjrZaBpGm0y6rX1Wr2H7E3KE6MDX4Co+jKLgajYJZGlBjYkApDFVVoRiYeIVkYnwEzYytSMERiZBuQcG9ssmwn75dLJqhcFO3konxlLvETjmJMYpiX5csKrvr6qeaEebPqqhKISBmknzGL9rL52r0RtG6Hn5pgY0IqmIR4K5QsNAlyWAzVMcowapR0q6JJ/i7hlBZg7xVNZQfnJegppkqm+fzUjdI/1wOKilikzVAiMaaP3d78ZBZ/HR4r8F6P08+bKLsxQp948TcOc53i5xaklferVvKKxueY0xrNDaILVd7I7kEGxm3uT0F2+pBtCLg5BdntQaaEeTAJ2etBxiYGOgkZ9SD7VmCGaAqy34PsYs9DwRTkoAfZZo7vTkIOe5AtbJv28HhuKygFNk5AL+XrYqEeXKgYOPYxN2r/DIeszN2yxVEksHKJccp9rfriXS9YxJ+vVDWp42i9UTwN8iummAv3apM8CVdNNjK+OK/hwldfjhPeGX8oZzq6nel8IGfaup3pfiBnWrqd2ZcMuYfUv4Zk+HsYJobTx9/EKPkfx5faXx/Bx/Gl9rwWfhxf6plwtSe1IsXJGsYuDLlaGE5asPQtDDkAoZMm/30LQ06YEzAdC0OOMPMmZSyvd5kchA7SkAs/MYaIPWkE+xaG1PSQbelYGIbUtach9y0MfeK4FkxeGD5DHIufx2uI69IQp4Rr0ZBSfRDq0RCYHmjREKWhS7RoCIVqt0mHhojvhxbXoiHfwxbSoiHXtS1zsobWKUAyVkKmkpCrQ0IUccQ8HRIKMSDi6pAQCpgTWjok5AaE+IEOCTmBisZQh4RsHnge1rI/6SPfDXVIyLY912GTJZQBjkcqCFA+ldOhIGJzD7AOBQUWIKYlCbmcWVTLRM4JiI2pFgVZoR96WhREfewTLQoKPEBaJnK244JnT1YQef3DaK+CCA8416IgAIdrWQqFJmVcTw5yiANMSw6ywxAsPQoKbKplGmcTRImpRUHUswNbi4Kw64zYfm0rKEpYtBbjNeTq0RB1GOagQ0PYIRMzZ5+GfB5SriULIeTDtJzcpyEPI58FOjTkUs8iWpZCrlrzh6YODTnUQYhM1tBun+7i0VtymHt6NAQhCzjToSG1B0X1aIj4oTstw/VpKOQ+1qOhwPQw0zKTUwtORLVsyXmeQ0OsQ0Ou6zg+nq6hKHkcqyCuFOTrUBAH5oOtQ0GcqCkG6FAQR4HaFdWzqY1crGdT21MTZqpDQSRw7EDLTC70bYSYDgX5pu164VgF5R+n044iP2Gb1X/RWs1eb3TXj6Waf+k4lHiGagv11iFvbzzcJlpvugHDwYB1N3CRZN/jbRRH0Dz6uQHMmgftW3z/SjJIOxibqG07iGDP3bifBRGZGANctyeOEvgB8gMhI82xuse3gTnOGMsdhdqwBDLl4C9X7jsw4HgfZ92sXSdyW5BtU6KEpoAl3Ihaqztse8AZ9ILPvXHYdW/tcIrXKd5trjlsY3WTOl3XkF4jvtKj3Q1ro7GwbVX+2B3Fn2G9j3F6Q0OnBp0i+rErniuKv4v4xkiVtePBc7Ho7HyO/649bw/Kl+iX9nvyRmy9nB9ZbOz8ApS1sL0b7BX6mDirUzhvoHCGUbhvoHCHUXhvoPCGUaA3UKBBFHkgFiTjGXoSChqPePMW2+s+m4tggmNuXsK7RoKGk9RVCAe8zS+yyA1mjelajbHWi8O1jPhcFjYvQ6n98fQ002tUyJ1KXOwaTjXNzCvSNcFfm3fF/8L9psO0KElUGqt63vZg/a7Rod5I2X+c3c1SWEcyT4Pffb9PaH7fSC4e5Ox+Fm13Is2ML9lzDN9FRQ1Onw2eiq3xlcxL5+xc/NVfV0n1wItxgjSOVeu/lUfA5tUlI8nmWYoTyUW6lfmTq2S5NBqld4adMCPPalvIkVShIYrbULK4BUXAwIyptJ8/+QXAyO8yXW42raNssycLKrbLG9Sr5NTTr1tWfqM6pXxDRcKjdetWshT7tDxP9t+zFC63l0vf/nSabYnmlOEJUiJktUCoTbkac44UdqmgIGV1b+pM0zClSbSLcVbaVKeTVLb1c7b+n6nY78putHDvrrSud6SsStTILB+BYHJZEf1UD/x9FLN/42xTBV/+dZl3aNlg4FEMbfDa+qyc48ksSvDlKtz/nnAa5ZfU5KK08K75QN5hnJVtC8Zz+/rhv+PdFMY5iQV9HM/balrFcms1WofJ2uuKS4N8MgXVG2L57Z9Wq+tHI1X0b0G99PLQria8onFptUapRuGHAvUfHQ81n+k9NFmPhOu23rRzgI0D7RtnW/ehz4e3CemqGh7eQQsXKTzcDsgH/ITLK6VLkOiq0b/OoMmNyhrvO2rn9Hb8Pw==)!?

**Comment by DarioSoller** [comment]
I do not want to annoy you with some genAI crap, but I tried to let it create a refactor plan for a `formatting.fileHeader` options API. In the following what it suggested. The only thing I am not sure about, if it is really necessary to rename `commentStyle` to be `tokenCommentStyle`!?

# Plan: Refactor File Header Settings into formatting.fileHeader

The current implementation has a critical architectural bug: [lib/common/formatHelpers/fileHeader.js](lib/common/formatHelpers/fileHeader.js) imports the wrong enum (`commentStyles` instead of `fileHeaderCommentStyles`), forcing workarounds with string literals. While the two enums serve conceptually distinct purposes (file headers vs token comments), they're incorrectly mixed. This plan separates concerns cleanly with dedicated `formatting.fileHeader` object, fixes enum usage, renames `commentStyles` to `tokenCommentStyles`, and provides full backward compatibility.

## Steps

### 1. Create new type structure in [types/File.ts](types/File.ts#L21-L32)

- Add `FileHeaderFormatting` interface with: `commentStyle?: 'short' | 'long' | 'xml'`, `prefix?`, `lineSeparator?`, `header?`, `footer?`, `timestamp?: boolean`
- Update `FormattingOverrides` to include `fileHeader?: FileHeaderFormatting`
- Add JSDoc `@deprecated` to `fileHeaderTimestamp` in `FormattingOverrides` (moved to `fileHeader.timestamp`)
- Keep existing `commentStyle?: commentStyles[keyof commentStyles]` for token comments with JSDoc note: "For token comments. For file header comment style, use `formatting.fileHeader.commentStyle`"
- Import and reference both enum types appropriately in TypeScript definitions

### 2. Rename commentStyles enum in [lib/enums/commentStyles.js](lib/enums/commentStyles.js)

- Rename primary export from `commentStyles` to `tokenCommentStyles` (values: `short`, `long`, `none`)
- Add backward compatible re-export: `export const commentStyles = tokenCommentStyles` with JSDoc `@deprecated Use tokenCommentStyles for token comments, or fileHeaderCommentStyles for file headers`
- Update [lib/enums/index.js](lib/enums/index.js#L1-L14) to export `tokenCommentStyles` as primary, `commentStyles` as deprecated alias
- Add documentation to [lib/enums/fileHeaderCommentStyles.js](lib/enums/fileHeaderCommentStyles.js) clarifying distinction: "Use for file header comments. Values: short, long, xml. For token property comments, use tokenCommentStyles"

### 3. Fix critical bug and add backward compatibility in [lib/common/formatHelpers/fileHeader.js](lib/common/formatHelpers/fileHeader.js#L1-L90)

- Change import from `commentStyles` to `fileHeaderCommentStyles` (line 1) - **fixes the existing bug**
- Update string literal `'xml'` check (line 82) to use `fileHeaderCommentStyles.xml`
- Implement backward compatibility layer: detect old patterns (`commentStyle` parameter, `formatting.fileHeaderTimestamp`, `file.options.commentStyle`) and map to `formatting.fileHeader`
- Add deprecation warning via `GroupMessages.add(FILE_HEADER_DEPRECATION, ...)` when old patterns detected
- New priority chain: `formatting?.fileHeader` > `file?.options?.formatting?.fileHeader` > legacy fallbacks (with warnings) > defaults
- Create `FILE_HEADER_DEPRECATION` constant in [lib/utils/groupMessages.js](lib/utils/groupMessages.js#L1-L19)

### 4. Refactor all format implementations in [lib/common/formats.js](lib/common/formats.js#L165-L1748)

- Replace `getFormattingCloneWithoutPrefix()` (line 165) with new `buildFileHeaderFormatting(commentStyle, formatting)` helper that constructs proper `fileHeader` object
- Update all 37 formats to use new helper: pass format-specific comment style ('short', 'long', or 'xml') and user's formatting to build merged config
- Remove all hardcoded `commentStyle` parameters from `fileHeader()` calls (lines 261, 295, 332, 365, 387, 419, 442, 667, 932, 967, 1002, 1037, 1074, 1110, 1210, and iOS formats)
- Update [lib/common/templates/android/resourceType.template.js](lib/common/templates/android/resourceType.template.js) and [lib/common/templates/ios/macros.template.js](lib/common/templates/ios/macros.template.js) similarly
- Ensure Android XML formats (lines 932, 967, 1002, 1037, 1074, 1110) use `commentStyle: 'xml'`, iOS plist (line 1210) uses `'xml'`, others use appropriate values

### 5. Update token comment usages to new enum in [lib/common/formatHelpers/formattedVariables.js](lib/common/formatHelpers/formattedVariables.js) and [lib/common/formatHelpers/createPropertyFormatter.js](lib/common/formatHelpers/createPropertyFormatter.js)

- Change imports from `commentStyles` to `tokenCommentStyles` (keep backward compatible alias working)
- Update destructuring: `const { short, long, none } = tokenCommentStyles`
- Update default value usage (line 247 in formattedVariables.js, line 31 in createPropertyFormatter.js)
- No breaking changes needed since values are identical, just clearer naming

### 6. Update tests with new API and comprehensive deprecation coverage

- Add backward compatibility tests in [\_\_tests\_\_/common/formatHelpers/fileHeader.test.js](\_\_tests\_\_/common/formatHelpers/fileHeader.test.js): verify old `commentStyle` parameter works, verify `formatting.fileHeaderTimestamp` works, verify deprecation warnings emitted using `stubMethod(console, 'log')` pattern
- Update [\_\_integration\_\_/customFileHeader.test.js](\_\_integration\_\_/customFileHeader.test.js) and [\_\_integration\_\_/showFileHeader.test.js](\_\_integration\_\_/showFileHeader.test.js) to include both old and new API examples
- Update [\_\_tests\_\_/common/formatHelpers/formattedVariables.test.js](\_\_tests\_\_/common/formatHelpers/formattedVariables.test.js) to use `tokenCommentStyles` enum (verify backward compatible `commentStyles` alias still works)
- Regenerate snapshots in `__integration__/__snapshots__/` where needed
- Add snapshot test for deprecation warning message format in logging tests

### 7. Update documentation with migration guide

- Add comprehensive section in [docs/src/content/docs/reference/Utils/format-helpers.md](docs/src/content/docs/reference/Utils/format-helpers.md) documenting new `formatting.fileHeader` API
- Create migration guide showing transformation examples: Old `formatting: { fileHeaderTimestamp: true }` → New `formatting: { fileHeader: { timestamp: true } }`
- Update enum documentation explaining `tokenCommentStyles` (for token property comments) vs `fileHeaderCommentStyles` (for file header comments)
- Add [CHANGELOG.md](CHANGELOG.md) entries: "Deprecated" section for old patterns, "Added" section for new `formatting.fileHeader` API, note v6.0.0 removal
- Prominently note in migration guide: custom formats passing `commentStyle` parameter to `fileHeader()` should migrate to `formatting.fileHeader.commentStyle`

## Further Considerations

1. **Deprecation warning verbosity** - Show at `default` level since this affects user configurations directly, or only at `verbose` to reduce noise? Recommend `default` with helpful migration message.
2. **Fix the xml string literal immediately** - Should we fix the bug in [fileHeader.js](lib/common/formatHelpers/fileHeader.js#L82) (using wrong enum) as a patch release before this refactor, or include it as part of this change? Recommend including in this refactor to avoid two breaking changes.
3. **Platform-level formatting.fileHeader** - The research shows formats define file header styles, but should platform `options.formatting.fileHeader` override format defaults? Recommend yes for consistency with current `formatting` override behavior.

**Comment by jorenbroekema** [comment]
Well, renaming the two things because they have distinct usages is not the right solution imo, because they do describe the same concept. Therefore, just `commentStyle` for both usages is correct, but it should be configured in SD in a way where you can be distinct in where you use what.

```js
{
  platforms: {
    css: {
      files: [{
        format: 'css/variables',
        options: {
          formatting: {
            commentStyle: 'long', // override for this particular file in the css platform
          }
        }
      }],
      options: {
        formatting: {
          commentStyle: 'short', // for tokens, fileHeaders will follow it if a fileHeader commentStyle is not defined
          fileHeader: {
            commentStyle: 'long', // here? override for fileheaders specifically..
          }
        },
        fileHeader: {
          commentStyle: 'long', // or here? instead of wrapped inside "formatting".. not sure, all fileHeader options are related 
                                // to how the fileHeader is formatted, so maybe wrapping it inside that is cleaner
        }
      }
    }
  }
}
```

So:
- options on the file level always override those defined in the platform level because they are more granular, I think this works in the current situation ✅ 
- fileHeader + util (at least our built-in ones) follows the formatting.commentStyle from the options, for consistency this seems like a reasonable default behavior ✅ 
- fileHeader + util (ditto^) will use fileHeader.commentStyle if defined because that's more granular 🚧 
- we use commentStyle as an example but same thing should be done for other options that the token formatting and fileHeader formatting share, which is probably not the current behavior as I only recently fixed this for commentStyle. 🚧 


Let me know what you think 👍🏻 @DarioSoller

**Comment by jorenbroekema** [comment]
Also, it seems like some of the SCSS formats hardcode the commentStyle to the fileheader helper util, which can be seen here for example https://github.com/style-dictionary/style-dictionary/blob/main/lib/common/formats.js#L334

I don't know if it's wise to hardcode it and prevent overriding from the user config. I suppose in CSS "short" comment style is actually invalid. I don't know for SCSS/SASS coz I never use it but by the code I'm guessing long comment style is either uncommon or invalid?

**Comment by DarioSoller** [comment]
@jorenbroekema `long` or so called [block comments are fine for SCSS](https://sass-lang.com/documentation/syntax/comments/#in-scss). Only exception are inline comments for CSS, that's right.

> options on the file level always override those defined in the platform level because they are more granular, I think this works in the current situation

True but not for SCSS, because of the mentioned hardcoded commentStyle in the format.

Besides that, I am with you regarding consistently naming it `commentStyle` but it being dependent on where in the options it has been set.
